lib/src/main/java/uk/gov/di/ipv/core/library/domain/SharedAttributesDeserializer.java

```diff
@@ -28,43 +29,49 @@ protected SharedAttributesDeserializer(Class<?> vc) {
     @Override
     public SharedAttributes deserialize(JsonParser jsonParser, DeserializationContext ctxt)
             throws IOException {
-
         SharedAttributes.Builder sharedAttributesBuilder = new SharedAttributes.Builder();
 
         JsonNode node = jsonParser.getCodec().readTree(jsonParser);
-
-        JsonNode attributes = node.get("attributes");
-        if (attributes == null) {
+        if (node.isEmpty()) {
             return SharedAttributes.empty();
         }
 
-        JsonNode names = attributes.get("names");
-        if (names != null) {
-            List<String> givenNames = new ArrayList<>();
-            for (JsonNode name : names.get("givenNames")) {
-                givenNames.add(name.asText());
-            }
-            JsonNode familyName = names.get("familyName");
-            if (familyName != null) {
-                sharedAttributesBuilder.setName(new Name(givenNames, familyName.asText()));
+        JsonNode nameNode = node.get("name");
+        if (nameNode != null) {
+            Set<Name> nameList = new HashSet<>();
+            List<NameParts> namePartsList = new ArrayList<>();
+
+            for (JsonNode jo : nameNode) {
+                JsonNode nameParts = jo.get("nameParts");
+                if (nameParts != null) {
+                    nameParts.forEach(
+                            namePart ->
+                                    namePartsList.add(
+                                            objectMapper.convertValue(namePart, NameParts.class)));
+                }
             }
+
+            Name names = new Name(namePartsList);
+            nameList.add(names);
+            sharedAttributesBuilder.setName(nameList);
```

Comment:
A name node is a list of objects. Each object has one key which is `nameParts`.
I feel like this may not handle the situation where the `name` list has more than one object. I think it will add all objects under `nameParts` to one `Name` object, rather than building individual `namePartsList`s for each entry in the `name` list.

Reply:
this is correct . The list of nameparts are added to the Name object (also shown correctly in the test)